### PR TITLE
pdm lock --check pre-commit hook

### LINKED ISSUE #1471: pdm check: check if the lockfile is up to date and related pre-commit hook
## Is your feature request related to a problem? Please describe.

I'd like to be able to guarentee that the lockfile is up to date when I update pyproject.toml. Because pdm stores a hash of specific keys found in pyproject.toml within pdm.lock, I want a command that simply exposes that information.

## Describe the solution you'd like
Simply put, `pdm check` which errors with a nonzero error code if the hash stored in pdm.lock doesn't match pyproject.toml or pdm.lock does not exist.

Furthermore, this could be exposed as a pre-commit hook which would check if the hash matches when pyproject.toml is edited.

### PR DESCRIPTION
**Pull request by onerandomusername**
## Pull Request Check List

- [x] A news fragment is added in `news/` describing what is new.
- [x] Test cases added for changed code.

## Describe what you have changed in this PR.

closes #1471 :)

### CODE REVIEW

**Comment by onerandomusername** [comment]
seems like pre-commit is failing for code that wasn't changed in this pull

File: .pre-commit-hooks.yaml [hunk 1]
```diff
@@ -1,3 +1,11 @@
+- id: pdm-check
+  name: pdm-check
```
Comment (frostming) [inline .pre-commit-hooks.yaml:2]: Can you make the name `pdm-lock-check` to be more specific?

**Review by frostming — CHANGES_REQUESTED** [review]

File: docs/docs/usage/advanced.md [hunk 2]
```diff
@@ -240,3 +240,15 @@ This hook wraps the command `pdm export` along with any valid argument. It can b
       args: ['-o', 'requirements.txt', '--without-hashes']
       files: ^pdm.lock$
 ```
+
+### Check `.pdm.lock` is up to date
```
Comment (frostming) [inline docs/docs/usage/advanced.md:244]:
```suggestion
### Check `pdm.lock` is up to date
```

File: docs/docs/usage/advanced.md [hunk 3]
```diff
@@ -240,3 +240,15 @@ This hook wraps the command `pdm export` along with any valid argument. It can b
...
+
+This hook wraps the command `pdm lock --check` along with any valid argument. It can be used as a hook (e.g., for CI) to ensure that whenever `pyproject.toml` has a dependency added/changed/removed, that `.pdm.lock` is also up to date.
+
+```yaml
+# export python requirements
```
Comment (frostming) [inline docs/docs/usage/advanced.md:249]: This line seems copied from other section, need change

File: docs/docs/usage/advanced.md (hunk 2 above)
Comment (onerandomusername) [reply docs/docs/usage/advanced.md]: https://github.com/pdm-project/pdm/pull/1475/commits/b0e5e824297ea5d1b14760d1d2d00e90a0e50c6d

File: docs/docs/usage/advanced.md (hunk 3 above)
Comment (onerandomusername) [reply docs/docs/usage/advanced.md]: https://github.com/pdm-project/pdm/pull/1475/commits/70809b43c16dae58f970a5af6057044a01e19c7a

File: .pre-commit-hooks.yaml (hunk 1 above)
Comment (onerandomusername) [reply .pre-commit-hooks.yaml]: https://github.com/pdm-project/pdm/pull/1475/commits/b0e5e824297ea5d1b14760d1d2d00e90a0e50c6d

**Review by frostming — APPROVED** [review]

**Pull request merged**